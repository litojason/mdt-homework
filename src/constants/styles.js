const colors = {
  primary: "#E72934",
  secondary: "#000",
  background: "#F5F5F5",
  modal: "rgba(0, 0, 0, 0.5)",
  cardBg: "#FFF",
  text: "#000",
  buttonText: "#FFF",
  shadowColor: "#000",
  red: "#cc3300",
  green: "#99cc33",
};

const spacing = {
  base: "1rem",
  half: "0.5rem",
  quarter: "0.25rem",
};

const borderRadius = {
  small: "0.25rem",
  base: "0.5rem",
  large: "1rem",
};

const borderWidth = {
  thin: "1px",
  base: "2px",
  thick: "3px",
};

const shadow = {
  base: "0px 6px 12px 0px rgba(0, 0, 0, 0.125)",
  //   card: "0px 6px 12px 0px rgba(0, 0, 0, 0.125)",
  card: "0px 3px 6px 0px rgba(0, 0, 0, 0.125)",
  //   navbar: "0px 6px 12px 0px rgba(0, 0, 0, 0.125)",
  navbar: "0px 3px 6px 0px rgba(0, 0, 0, 0.125)",
};

const fonts = {
  size: {
    base: "1rem",
    description: "14px",
    title: "1.5rem",
  },
};

const icons = {
  size: {
    base: "1.5rem",
  },
};

export { colors, spacing, borderRadius, borderWidth, shadow, fonts, icons };
