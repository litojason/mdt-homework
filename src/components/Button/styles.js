import styled from "styled-components";
import { AiOutlineLoading } from "react-icons/ai";

import { spacing, colors, borderRadius } from "constants/styles";

export const Button = styled.button`
  padding: ${spacing.base};
  background-color: ${colors.primary};
  border: 0;
  color: ${colors.buttonText};
  border-radius: ${borderRadius.base};
  opacity: ${({ loading }) => (loading === "true" ? 0.8 : 1)};
`;

export const LoadingSpinner = styled(AiOutlineLoading)`
  animation: rotation 0.5s infinite linear;

  @keyframes rotation {
    from {
      transform: rotate(0deg);
    }
    to {
      transform: rotate(359deg);
    }
  }
`;
