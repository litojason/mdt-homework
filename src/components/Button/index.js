import React from "react";
import PropTypes from "prop-types";

import { Button as StyledButton, LoadingSpinner } from "./styles";

const Button = ({ type, onClick, loading, children }) => {
  return (
    <StyledButton
      type={type}
      onClick={onClick}
      loading={loading ? loading.toString() : "false"}
      disabled={loading}
    >
      {loading ? <LoadingSpinner /> : children}
    </StyledButton>
  );
};

Button.propTypes = {
  type: PropTypes.string,
  onClick: PropTypes.func,
  loading: PropTypes.bool,
  children: PropTypes.element,
};

export default Button;
