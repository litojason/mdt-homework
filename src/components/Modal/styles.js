import styled from "styled-components";

import { borderRadius, colors, fonts, shadow, spacing } from "constants/styles";

export const Background = styled.div`
  position: fixed;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100vw;
  height: 100vh;
  padding: ${spacing.base};
  background-color: ${colors.modal};
  box-sizing: border-box;
  z-index: 20;
`;

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  padding: ${spacing.base};
  border-radius: ${borderRadius.base};
  background-color: ${colors.cardBg};
  box-shadow: ${shadow.card};
`;

export const Title = styled.div`
  font-size: ${fonts.size.title};
  font-weight: 600;
`;

export const Description = styled.div`
  margin: ${spacing.half} 0 ${spacing.base} 0;
  text-align: center;
`;
