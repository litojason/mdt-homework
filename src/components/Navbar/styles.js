import styled from "styled-components";
import { AiOutlineLeft } from "react-icons/ai";

import { colors, fonts, icons, shadow, spacing } from "constants/styles";

export const Nav = styled.nav`
  position: sticky;
  top: 0;
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100vw;
  height: 3rem;
  padding: ${spacing.base};
  background-color: ${colors.background};
  box-sizing: border-box;
  box-shadow: ${shadow.navbar};
`;

export const Title = styled.div`
  font-size: ${fonts.size.title};
`;

export const BackIcon = styled(AiOutlineLeft).attrs((props) => ({
  size: icons.size.base,
}))``;
