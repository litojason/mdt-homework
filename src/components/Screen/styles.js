import styled from "styled-components";

import { colors, spacing } from "constants/styles";

export const Screen = styled.div`
  display: flex;
  flex-direction: column;
  height: 100vh;
  box-sizing: border-box;
  background-color: ${colors.background};
`;
export const Content = styled.div`
  display: flex;
  flex-direction: column;
  height: 100%;
  padding: ${spacing.base};
`;
export const AuthFooter = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: ${spacing.half};
`;
