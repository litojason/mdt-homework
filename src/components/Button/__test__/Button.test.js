import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";

import Button from "../index";
import { LoadingSpinner } from "../styles";

describe("Component: Button", () => {
  it("should render Button", () => {
    const component = <Button />;

    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render Button loading", () => {
    const component = <Button loading={true} />;

    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render LoadingSpinner", () => {
    const component = <LoadingSpinner />;

    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
