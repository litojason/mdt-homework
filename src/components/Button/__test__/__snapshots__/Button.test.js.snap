// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Button should render 1`] = `
.c0 {
  padding: 1rem;
  background-color: #E72934;
  border: 0;
  color: #FFF;
  border-radius: 0.5rem;
  opacity: 1;
}

<button
  className="c0"
  loading="false"
/>
`;

exports[`Component: Button should render Button 1`] = `
.c0 {
  padding: 1rem;
  background-color: #E72934;
  border: 0;
  color: #FFF;
  border-radius: 0.5rem;
  opacity: 1;
}

<button
  className="c0"
  loading="false"
/>
`;

exports[`Component: Button should render Button loading 1`] = `
.c0 {
  padding: 1rem;
  background-color: #E72934;
  border: 0;
  color: #FFF;
  border-radius: 0.5rem;
  opacity: 0.8;
}

.c1 {
  -webkit-animation: rotation 0.5s infinite linear;
  animation: rotation 0.5s infinite linear;
}

<button
  className="c0"
  disabled={true}
  loading="true"
>
  <svg
    className="c1"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 1024 1024"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 0 0-94.3-139.9 437.71 437.71 0 0 0-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
    />
  </svg>
</button>
`;

exports[`Component: Button should render LoadingSpinner 1`] = `
.c0 {
  -webkit-animation: rotation 0.5s infinite linear;
  animation: rotation 0.5s infinite linear;
}

<svg
  className="c0"
  fill="currentColor"
  height="1em"
  stroke="currentColor"
  strokeWidth="0"
  style={
    Object {
      "color": undefined,
    }
  }
  viewBox="0 0 1024 1024"
  width="1em"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 0 0-94.3-139.9 437.71 437.71 0 0 0-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
  />
</svg>
`;
